airflow/contrib/hooks/kubernetes_hook.py

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import tempfile
+
+from kubernetes import client, config
+
+from airflow.hooks.base_hook import BaseHook
+
+
+class Kuberneteshook(BaseHook):
+    """
+    creates kubernetes API connection.
+    :param conn_id: the connection to Kubernetes cluster
+    """
+
+    def __init__(
+        self,
+        conn_id="kubernetes_default"
+    ):
+        self.connection = self.get_connection(conn_id)
+
+    def get_conn(self):
+        """
+        Returns kubernetes api session for use with requests
+        """
+        if self.connection.extra is None or self.connection.extra == '':
+            self.log.info("loading kube_config from: default file")
+        else:
+            extra_obj = self.connection.extra_dejson
+            if "in_cluster" in extra_obj.keys():
```

Comment:
All keys should have a crazy prefix, otherwise it will not be compatible with our Web UI. You will not be able to add the edit box in the connection edit form. 
For example:
GCP:
https://airflow.readthedocs.io/en/latest/howto/connection/gcp.html - `extra__google_cloud_platform__`
GRPC:
`extra__grpc__`
https://github.com/apache/airflow/blob/master/airflow/contrib/hooks/grpc_hook.py#L123-L134
You can copy the get_field method if you need them.